pkg/apis/meta/v1/unstructured/converter.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unstructured
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/golang/glog"
+)
+
+// Converter knows how to convert betweek runtime.Object and
+// Unstructured in both ways.
+type Converter struct {
+}
+
+func NewConverter() *Converter {
+	return &Converter{}
+}
+
+func (c *Converter) FromUnstructured(u *Unstructured, obj runtime.Object) error {
```

Comment:
Use map[string]interface{} instead of Unstructured - the way the code is currently factored, "Unstructured" means "tied to meta/v1 schema" but nothing in this code is tied that way.  At some point "Unstructured" will become a struct wrapper (`type Foo struct { Unstructured }`) that adds the appropriate accessors for type comparison.